### PR TITLE
chore(statsig): get QR code experiment part 2 params from statsig

### PR DESCRIPTION
**Pull request by cajubelt**
### Description

uses parameters from Statsig layer to determine whether to show new or old QR code style, and whether to encode a valora deep link or just an address string in the QR code. Affects QR codes reachable from the send/request flow, not the exchange flow, which is in a separate experiment ("part 1" of the qr experiment the activation squad is running)

### Test plan

added automated unit and integration tests and tested manually

https://user-images.githubusercontent.com/7979215/216184909-60d0c005-9b25-40b0-ba20-f90185eb5c89.mp4

https://user-images.githubusercontent.com/7979215/216165031-dd196a4a-5a8f-41ae-991f-2ed4c02c681a.mp4

### Related issues

- Fixes https://linear.app/valora/issue/ACT-551/setup-statsig-experiment-and-use-params-in-wallet

### Backwards compatibility

backwards compatible

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -178,7 +178,7 @@
     "redux-persist-fs-storage": "^1.3.0",
     "redux-saga": "^1.1.3",
     "reselect": "^4.1.7",
-    "statsig-react-native": "^4.6.0",
+    "statsig-react-native": "^4.9.2",
```
Comment (cajubelt) [inline package.json:181]: needed this to use the statsig layer override api for testing

**Review by satish-ravi — APPROVED** [review]

File: src/navigator/QRNavigator.tsx [hunk 2]
```diff
@@ -45,10 +47,33 @@ type AnimatedScannerSceneProps = NativeStackScreenProps<QRTabParamList, Screens.
...
+  const { paramName: dataTypeParamName, defaultValue: dataTypeDefaultValue } =
+    LayerParams[layerName].qrCodeDataType
+  try {
+    const statsigLayer = Statsig.getLayer(layerName)
+    const qrCodeStyle = statsigLayer.get(styleParamName, styleDefaultValue)
+    const qrCodeDataType = statsigLayer.get(dataTypeParamName, dataTypeDefaultValue)
+    return { qrCodeStyle, qrCodeDataType }
+  } catch (error) {
+    Logger.warn(TAG, 'error getting Statsig experiment', error)
+    return {
```
Comment (jophish) [inline src/navigator/QRNavigator.tsx:66]: nit: test that this returns defaults on error?

File: src/navigator/QRNavigator.test.tsx [hunk 3]
```diff
@@ -46,6 +57,25 @@ function getProps(qrCodeStyle: QRCodeStyle): QRCodeProps {
...
 
 describe('QRNavigator', () => {
+  beforeAll(async () => {
+    await Statsig.initialize('client-sdk-key-testing', null, { localMode: true })
+  })
+  afterEach(async () => {
+    const overrides = Statsig.getAllOverrides()
+    Object.keys(overrides.gates).forEach((gateName) => Statsig.removeGateOverride(gateName))
+    Object.keys(overrides.configs).forEach((configName) => Statsig.removeConfigOverride(configName))
+    Object.keys(overrides.layers).forEach((layerName) => Statsig.removeLayerOverrie(layerName)) // lol @ the typo, I (Charlie) submitted a PR to fix their SDK but it will take time to get a new version published
```
Comment (jophish) [inline src/navigator/QRNavigator.test.tsx:67]: :facepalm:

**Review by jophish — COMMENTED** [review]
Looks fantastic; one potential suggestion for a test, but not really critical.

File: src/navigator/QRNavigator.test.tsx (hunk 3 above)
Comment (cajubelt) [reply src/navigator/QRNavigator.test.tsx:67]: they at least merged my PR

Comment (cajubelt) [reply src/navigator/QRNavigator.test.tsx:67]: I love that Statsig is open source so I can do stuff like that instead of just report and wait

Comment (cajubelt) [reply src/navigator/QRNavigator.test.tsx:67]: they don't have CD set up to publish new NPM versions though sadly

**Pull request merged**